### PR TITLE
Add missing examples

### PR DESCRIPTION
**Pull request by mbaudin47**
This PR creates new examples for the following PCE related classes:
- DesignProxy
- KFold
- LARS
- LeastSquaresMethod
- SparseMethod

### CODE REVIEW

File: python/src/LARS_doc.i.in [hunk 1]
```diff
@@ -24,5 +24,36 @@ The sparse least squares approaches may be employed instead. Eventually a sparse
...
+>>> # Create the orthogonal basis
+>>> polynomialCollection = ot.PolynomialFamilyCollection(im.dim)
+>>> for i in range(im.dim):
+...    polynomialCollection[i] = ot.OrthogonalUniVariatePolynomialFamily(
+...        ot.LegendreFactory()
+...    )
+>>> enumerateFunction = ot.LinearEnumerateFunction(im.dim)
+>>> productBasis = ot.OrthogonalBasis(
+...    ot.OrthogonalProductPolynomialFactory(polynomialCollection, enumerateFunction)
+...)
```
Comment (mbaudin47) [inline python/src/LARS_doc.i.in:41]:
```suggestion
... )
```

File: python/src/LARS_doc.i.in [hunk 2]
```diff
@@ -24,5 +24,36 @@ The sparse least squares approaches may be employed instead. Eventually a sparse
...
 
-This class is not usable as is because it has sense only whithin the
-:class:`~openturns.FunctionalChaosAlgorithm`."
+Examples
+--------
+>>> import openturns as ot
+>>> from openturns.usecases import ishigami_function
+>>> im = ishigami_function.IshigamiModel()
+>>> # Create the orthogonal basis
+>>> polynomialCollection = ot.PolynomialFamilyCollection(im.dim)
```
Comment (regislebrun) [inline python/src/LARS_doc.i.in:33]: You can use:
```
polynomialCollection = [ot.LegendreFactory()]*im.dim
```

File: python/src/LARS_doc.i.in [hunk 3]
```diff
@@ -24,5 +24,36 @@ The sparse least squares approaches may be employed instead. Eventually a sparse
...
+>>> for i in range(im.dim):
+...    polynomialCollection[i] = ot.OrthogonalUniVariatePolynomialFamily(
+...        ot.LegendreFactory()
+...    )
+>>> enumerateFunction = ot.LinearEnumerateFunction(im.dim)
+>>> productBasis = ot.OrthogonalBasis(
```
Comment (regislebrun) [inline python/src/LARS_doc.i.in:39]: No need to cast `ot.OrthogonalProductPolynomialFactory` into `ot.OrthogonalBasis`.
If you use the linear enumeration you don't even have to pass it to the constructor of `ot.OrthogonalProductPolynomialFactory` (but you may want to make it explicit)

File: python/src/LeastSquaresMethod_doc.i.in [hunk 4]
```diff
@@ -53,4 +53,26 @@ design : 2-d sequence of float
...
+
+Examples
+--------
+>>> import openturns as ot
+>>> basisSize = 3
+>>> sampleSize = 5
+>>> X = ot.Sample(sampleSize, 1)
+>>> for i in range(sampleSize):
+...     X[i, 0] = i + 1.0
+>>> phis = []
```
Comment (regislebrun) [inline python/src/LeastSquaresMethod_doc.i.in:66]:
```
phis = [ot.SymbolicFunction(['x'], ['x^' + str(j + 1)]) for j in range(basisSize)]
```

File: python/src/SparseMethod_doc.i.in [hunk 5]
```diff
@@ -17,4 +17,26 @@ fittingAlgorithm : :class:`~openturns.FittingAlgorithm`
...
+
+Examples
+--------
+>>> import openturns as ot
+>>> basisSize = 3
+>>> sampleSize = 5
+>>> X = ot.Sample(sampleSize, 1)
+>>> for i in range(sampleSize):
+...     X[i, 0] = i + 1.0
+>>> phis = []
```
Comment (regislebrun) [inline python/src/SparseMethod_doc.i.in:30]:
```
phis = [ot.SymbolicFunction(['x'], ['x^' + str(j + 1)]) for j in range(basisSize)]
```

**Review by regislebrun — APPROVED** [review]
Nice examples. Some casts are unnecessary, and some loops could be avoided but nothing blocking.

File: python/src/DesignProxy_doc.i.in [hunk 6]
```diff
@@ -20,6 +20,20 @@ Notes
...
 to select the algorithm used for the resolution of linear least-squares problems.
+
+Examples
+--------
+>>> import openturns as ot
+>>> basisSize = 3
+>>> sampleSize = 5
+>>> X = ot.Sample(sampleSize, 1)
+>>> for i in range(sampleSize):
+...     X[i, 0] = i + 1.0
```
Comment (josephmure) [inline python/src/DesignProxy_doc.i.in:31]:
```suggestion
>>> X = ot.Sample.BuildFromPoint(range(sampleSize))
```

File: python/src/KFold_doc.i.in [hunk 7]
```diff
@@ -15,8 +15,21 @@ Notes
...
+>>> import openturns as ot
+>>> size = 100
+>>> xuniform = ot.Uniform(0.9, 1.1)
+>>> x = xuniform.getSample(size)
+>>> yuniform = ot.Uniform(1.9, 2.1)
+>>> y = yuniform.getSample(size)
+>>> w = [1.0] * size
+>>> f = ot.SymbolicFunction(['x'], ['2.0 * x'])
+>>> basis = [f]
+>>> indices = list(range(len(basis)))
```
Comment (josephmure) [inline python/src/KFold_doc.i.in:29]:
```suggestion
>>> indices = [0]
```
I get you want to be generic but this is overkill to me...

File: python/src/LeastSquaresMethod_doc.i.in [hunk 8]
```diff
@@ -53,4 +53,26 @@ design : 2-d sequence of float
...
 method : :class:`~openturns.LeastSquaresMethod`
-    The built method"
+    The built method
+
+Examples
+--------
+>>> import openturns as ot
+>>> basisSize = 3
+>>> sampleSize = 5
+>>> X = ot.Sample(sampleSize, 1)
```
Comment (josephmure) [inline python/src/LeastSquaresMethod_doc.i.in:63]: Same comments as before.

File: python/src/LeastSquaresMethod_doc.i.in [hunk 9]
```diff
@@ -53,4 +53,26 @@ design : 2-d sequence of float
...
+...     phis.append(ot.SymbolicFunction(['x'], ['x^' + str(j + 1)]))
+>>> basis = ot.Basis(phis)
+>>> proxy = ot.DesignProxy(X, phis)
+>>> indices = range(basisSize)
+>>> designMatrix = ot.Matrix(proxy.computeDesign(indices))
+>>> method = ot.LeastSquaresMethod.Build('SVD', designMatrix)
+>>> y = ot.Normal(
+...     [1.0] * sampleSize,
+...     [0.1] * sampleSize,
+... ).getRealization()
```
Comment (josephmure) [inline python/src/LeastSquaresMethod_doc.i.in:77]:
```suggestion
>>> normal = ot.Normal([1.0] * sampleSize, [0.1] * sampleSize)
>>> y = normal.getRealization()
```

File: python/src/SparseMethod_doc.i.in [hunk 10]
```diff
@@ -17,4 +17,26 @@ fittingAlgorithm : :class:`~openturns.FittingAlgorithm`
...
 --------
-LeastSquaresMethod"
+LeastSquaresMethod
+
+Examples
+--------
+>>> import openturns as ot
+>>> basisSize = 3
+>>> sampleSize = 5
+>>> X = ot.Sample(sampleSize, 1)
```
Comment (josephmure) [inline python/src/SparseMethod_doc.i.in:27]: Same as before.

**Review by josephmure — CHANGES_REQUESTED** [review]

File: python/src/DesignProxy_doc.i.in (hunk 6 above)
Comment (mbaudin47) [inline python/src/DesignProxy_doc.i.in:31]: Not exactly. This corresponds exactly:
```suggestion
>>> X = ot.Sample.BuildFromPoint(range(1, 1 + sampleSize))
```
@josephmure : Do you agree?

File: python/src/LARS_doc.i.in [hunk 11]
```diff
@@ -24,5 +24,36 @@ The sparse least squares approaches may be employed instead. Eventually a sparse
...
+--------
+>>> import openturns as ot
+>>> from openturns.usecases import ishigami_function
+>>> im = ishigami_function.IshigamiModel()
+>>> # Create the orthogonal basis
+>>> polynomialCollection = ot.PolynomialFamilyCollection(im.dim)
+>>> for i in range(im.dim):
+...    polynomialCollection[i] = ot.OrthogonalUniVariatePolynomialFamily(
+...        ot.LegendreFactory()
+...    )
```
Comment (mbaudin47) [inline python/src/LARS_doc.i.in:37]:
```suggestion
>>> polynomialCollection = [ot.LegendreFactory()] * im.dim
```

File: python/src/LARS_doc.i.in (hunk 1 above)
Comment (mbaudin47) [inline python/src/LARS_doc.i.in:41]:
```suggestion
>>> productBasis = ot.OrthogonalProductPolynomialFactory(polynomialCollection, enumerateFunction)
```

File: python/src/LeastSquaresMethod_doc.i.in [hunk 12]
```diff
@@ -53,4 +53,26 @@ design : 2-d sequence of float
...
+    The built method
+
+Examples
+--------
+>>> import openturns as ot
+>>> basisSize = 3
+>>> sampleSize = 5
+>>> X = ot.Sample(sampleSize, 1)
+>>> for i in range(sampleSize):
+...     X[i, 0] = i + 1.0
```
Comment (mbaudin47) [inline python/src/LeastSquaresMethod_doc.i.in:65]:
```suggestion
>>> X = ot.Sample.BuildFromPoint(range(1, 1 + sampleSize))
```

File: python/src/LeastSquaresMethod_doc.i.in [hunk 13]
```diff
@@ -53,4 +53,26 @@ design : 2-d sequence of float
...
+>>> phis = []
+>>> for j in range(basisSize):
+...     phis.append(ot.SymbolicFunction(['x'], ['x^' + str(j + 1)]))
```
Comment (mbaudin47) [inline python/src/LeastSquaresMethod_doc.i.in:68]:
```suggestion
>>> phis = [ot.SymbolicFunction(['x'], ['x^' + str(j + 1)]) for j in range(basisSize)]
```

File: python/src/SparseMethod_doc.i.in [hunk 14]
```diff
@@ -17,4 +17,26 @@ fittingAlgorithm : :class:`~openturns.FittingAlgorithm`
...
+>>> for i in range(sampleSize):
+...     X[i, 0] = i + 1.0
```
Comment (mbaudin47) [inline python/src/SparseMethod_doc.i.in:29]:
```suggestion
>>> X = ot.Sample.BuildFromPoint(range(1, 1 + sampleSize))
```

File: python/src/SparseMethod_doc.i.in [hunk 15]
```diff
@@ -17,4 +17,26 @@ fittingAlgorithm : :class:`~openturns.FittingAlgorithm`
...
+>>> phis = []
+>>> for j in range(basisSize):
+...     phis.append(ot.SymbolicFunction(['x'], ['x^' + str(j + 1)]))
```
Comment (mbaudin47) [inline python/src/SparseMethod_doc.i.in:32]:
```suggestion
>>> phis = [ot.SymbolicFunction(['x'], ['x^' + str(j + 1)]) for j in range(basisSize)]
```

File: python/src/SparseMethod_doc.i.in [hunk 16]
```diff
@@ -17,4 +17,22 @@ fittingAlgorithm : :class:`~openturns.FittingAlgorithm`
...
+>>> X = ot.Sample.BuildFromPoint(range(1, 1 + sampleSize))
+>>> phis = [ot.SymbolicFunction(['x'], ['x^' + str(j + 1)]) for j in range(basisSize)]
+>>> basis = ot.Basis(phis)
+>>> proxy = ot.DesignProxy(X, phis)
+>>> full = range(basisSize)
+>>> design = ot.Matrix(proxy.computeDesign(full))
+>>> method = ot.SparseMethod(ot.QRMethod(proxy, full))
+>>> y = ot.Normal(
+...     [1.0] * sampleSize, [0.1] * sampleSize,
+... ).getRealization()
```
Comment (josephmure) [inline python/src/SparseMethod_doc.i.in:36]:
```suggestion
>>> normal = ot.Normal([1.0] * sampleSize, [0.1] * sampleSize)
>>> y = normal.getRealization()
```

**Review by josephmure — APPROVED** [review]
OK for me after this change.

**Pull request merged**